packages/cli/src/lint/lint.ts

```diff
@@ -48,21 +48,28 @@ export async function lint(options: Options, config: ProjectConfig) {
   }
 
   const rules = lintLib.registry.getRules(ruleCodes);
-  const filter = new WarningFilter({
-    warningCodesToIgnore: new Set(lintOptions.ignoreWarnings || []),
-    minimumSeverity: Severity.WARNING,
-    filesToIgnore: lintOptions.filesToIgnore,
-  });
-
-  const {analyzer, urlLoader, urlResolver} = getConfiguredAnalyzer(config);
+  const {analyzer, urlLoader, urlResolver, warningFilter} =
+      await config.initializeAnalyzer();
   const linter = new lintLib.Linter(rules, analyzer);
 
   if (options.watch) {
     return watchLoop(
-        analyzer, urlLoader, urlResolver, linter, options, config, filter);
+        analyzer,
+        urlLoader,
+        urlResolver,
+        linter,
+        options,
+        config,
```

Comment:
Noticed that we pass around `config` but we only use `config.root`, would it make sense to refactor all these signatures to get the root instead?

Reply:
Next time I touch this file I'm gonna refactor this into a class rather than pass these same 7-8 options around everywhere